### PR TITLE
`Development`: Move course score calculation to the server

### PR DESCRIPTION
**Pull request by frederik-kt**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [ ] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [ ] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [ ] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [ ] I implemented the changes with a good performance and prevented too many database calls.
- [ ] I documented the Java code using JavaDoc style.
#### Client
- [ ] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [ ] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [ ] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] I documented the TypeScript code using JSDoc style.
- [ ] I added multiple screenshots/screencasts of my UI changes.
- [ ] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis
2. Navigate to Course Administration
3. ...

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exam with a Programming Exercise

1. Log in to Artemis
2. Participate in the exam as a student
3. Make sure that the UI of the programming exercise in the exam mode stays unchanged. You can use the [exam mode documentation](https://docs.artemis.ase.in.tum.de/user/exam_mode/) as reference.
4. ...

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java [hunk 1]
```diff
@@ -406,27 +406,25 @@ private ExamBonusCalculator createExamBonusCalculator(Optional<GradingScale> gra
...
             return new BonusResultDTO(bonusStrategy, bonusFromTitle, bonusExample.studentPointsOfBonusSource(), bonusGrade, bonusExample.finalPoints(), bonusExample.finalGrade(),
-                    verdict, achievedPresentationScore, presentationScoreThreshold);
+                verdict, achievedPresentationScore, presentationScoreThreshold);
         };
     }
 
     private Map<Long, BonusSourceResultDTO> calculateBonusSourceStudentPoints(GradingScale sourceGradingScale, Collection<Long> studentIds) {
         try {
             if (sourceGradingScale.getCourse() != null) {
-                CourseScoresDTO courseScores = courseScoreCalculationService.calculateCourseScores(sourceGradingScale.getCourse().getId(), studentIds);
```
Comment (frederik-kt) [inline src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java:422]: This is the only change in this file. The rest is just Intellij reformatting the code. If this is an issue please tell me what our policy is on that.

File: src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java [hunk 2]
```diff
@@ -403,34 +411,46 @@ public List<Course> getAllCoursesToRegister() {
      *
      * @param courseId the courseId for which exercises, lectures, exams and learning goals should be fetched
      * @return a course with all exercises, lectures, exams and learning goals visible to the student
+     * In addition, scores per exercise type and the participant scores for each exercise are sent back as an optimization.
      */
     @GetMapping("courses/{courseId}/for-dashboard")
     @PreAuthorize("hasRole('USER')")
-    public Course getCourseForDashboard(@PathVariable long courseId) {
```
Comment (frederik-kt) [inline src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java:409]: This method and the next one contain the important changes.

File: src/main/java/de/tum/in/www1/artemis/web/rest/dto/CourseScoresDTO.java [hunk 3]
```diff
@@ -1,27 +0,0 @@
-package de.tum.in.www1.artemis.web.rest.dto;
```
Comment (frederik-kt) [inline src/main/java/de/tum/in/www1/artemis/web/rest/dto/CourseScoresDTO.java:1]: Renamed to CourseScoresForExamBonusSourceDTO.

File: src/main/webapp/app/course/manage/course-storage.service.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
```
Comment (frederik-kt) [inline src/main/webapp/app/course/manage/course-storage.service.ts:1]: was in the course-score-calculation.service.ts before

**Comment by ge65cer** [comment]
Closes #2647

File: src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java (hunk 1 above)
Comment (b-fein) [reply src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java:422]: Usually `./gradlew spotlessApply` should trigger automatically as a pre-commit hook to format the Java files accordingly.
The Git hooks (that then also run the client code linting/formatter) should have been installed when running `npm install`.

**Pull request closed without merging**